File: src/main/java/com/pairingplanet/pairing_planet/dto/recipe/HomeFeedResponseDto.java
package com.pairingplanet.pairing_planet.dto.recipe;

import lombok.Builder;

import java.util.List;

@Builder
public record HomeFeedResponseDto(
        List<RecipeSummaryDto> recentRecipes,    // 최근 생성된 레시피
        List<TrendingTreeDto> trendingTrees      // "이 레시피, 이렇게 바뀌고 있어요" 섹션용
) {}

File: lib/data/models/home_feed_response_dto.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pairing_planet2_frontend/data/models/recipe/recipe_summary_dto.dart';
import 'recipe/trending_tree_dto.dart';

part 'home_feed_response_dto.g.dart';

@JsonSerializable()
class HomeFeedResponseDto {
  final List<RecipeSummaryDto> recentRecipes;
  final List<TrendingTreeDto> trendingTrees;

  HomeFeedResponseDto({
    required this.recentRecipes,
    required this.trendingTrees,
  });

  factory HomeFeedResponseDto.fromJson(Map<String, dynamic> json) =>
      _$HomeFeedResponseDtoFromJson(json);
  Map<String, dynamic> toJson() => _$HomeFeedResponseDtoToJson(this);
}
